tracer/build/_build/docker/universal.dockerfile

```diff
@@ -0,0 +1,94 @@
+﻿FROM datadog/libddwaf:toolchain as base
+
+ARG DOTNETSDK_VERSION
+
+# Based on https://github.com/dotnet/dotnet-docker/blob/34c81d5f9c8d56b36cc89da61702ccecbf00f249/src/sdk/6.0/bullseye-slim/amd64/Dockerfile
+# and https://github.com/dotnet/dotnet-docker/blob/1eab4cad6e2d42308bd93d3f0cc1f7511ac75882/src/sdk/5.0/buster-slim/amd64/Dockerfile
+ENV \
+    # Unset ASPNETCORE_URLS from aspnet base image
+    ASPNETCORE_URLS= \
+    # Do not generate certificate
+    DOTNET_GENERATE_ASPNET_CERTIFICATE=false \
+    # Do not show first run text
+    DOTNET_NOLOGO=true \
+    # Enable correct mode for dotnet watch (only mode supported in a container)
+    DOTNET_USE_POLLING_FILE_WATCHER=true \
+    # Skip extraction of XML docs - generally not useful within an image/container - helps performance
+    NUGET_XMLDOC_MODE=skip \
+    # Disable LTTng tracing with QUIC
+    QUIC_LTTng=0
+
+RUN apt-get update \
+    && apt-get -y upgrade \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --fix-missing \
+      cmake \
+        git \
+        procps \
+        wget \
+        curl \
+        cmake \
+        make \
+        gcc \
+        build-essential \
+        rpm \
+        ruby \
+        ruby-dev \
+        rubygems \
+        uuid-dev \
+        autoconf \
```

Comment:
Most of these are for doing packaging etc, but AFAICT we don't need any of that in this dockerfile. Could we slim it down to just the things we need? I'm kind of assuming datadog/libddwaf:toolchain already has all the non-dotnet things? 🤔

Reply:
yep, good point. This was amongst the things I wanted to clean out.
I wasn't sure what to remove.

Reply:
Based on the dockerfile here: 

https://github.com/DataDog/libddwaf/blob/445c1a8f2471c0461aeace0ae868924c8af01133/docker/libddwaf/toolchain/Dockerfile#L6

we already have 
- `cmake`
- `make`
- `git`
- `wget`
- `software-properties-common`

and I'm _fairly_ sure we don't need 

- `rpm`
- `ruby`
- `ruby-dev`
- `rubygems`
- `uuid-dev` (only used by profiler #2761)
- `autoconf` (only used by profiler #2761)
- `libtool` (only used by profiler #2761)
- `liblzma-dev` (only used by profiler #2761)
- `gdb` (only used by profiler #2761)

But I'm not 100% tbh. Will need to make sure these are cached on the VMs too (I can look into that once this is finalised)